provider/lib/details_screen.dart

```diff
@@ -0,0 +1,101 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:provider_sample/todo_list_model.dart';
+import 'package:todos_app_core/todos_app_core.dart';
+
+import 'edit_todo_screen.dart';
+import 'models.dart';
+
+class DetailsScreen extends StatelessWidget {
+  final String id;
+  final void Function() onRemove;
+
+  const DetailsScreen({@required this.id, @required this.onRemove})
+      : super(key: ArchSampleKeys.todoDetailsScreen);
+
+  @override
+  Widget build(BuildContext context) {
+    final model = Provider.of<TodoListModel>(context, listen: true);
+    final todo = model.todoById(id) ?? Todo('');
+
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(ArchSampleLocalizations.of(context).todoDetails),
+        actions: <Widget>[
+          IconButton(
+            key: ArchSampleKeys.deleteTodoButton,
+            tooltip: ArchSampleLocalizations.of(context).deleteTodo,
+            icon: const Icon(Icons.delete),
+            onPressed: onRemove,
+          )
+        ],
+      ),
+      floatingActionButton: FloatingActionButton(
+        key: ArchSampleKeys.editTodoFab,
+        onPressed: () {
+          Navigator.push(
```

Comment:
I like using `onGenerateRoute` in those situations.
The `onEdit` could be a method on `TodoListModel` once again.

It's not important though.